### PR TITLE
add makefiles for c/c++ with cppcheck

### LINKED ISSUE #16: Implement cppcheck run for C/C++  binaries
- [x] Have makefiles for C/C++ 
- [x] implement CPPcheck

### PR DESCRIPTION
**Pull request by diamant3**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [ ] A new challenge
-   [ ] Additional documentation
-   [x] Something else

#### Description
add makefiles for each c/c++ dir with cppcheck according to #16

### Relations

Closes #16

### Checklist:

-   [ ] All the contributions made are solely the work of me and my co-authors
-   [ ] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] Quickbuild.sh is extended to compile the binary for all target environnments
-   [ ] Github actions are implemented to publish the new challenge
-   [ ] A spoil command is implemented to return the actual answer
-   [ ] The correct and incorrect answer strings are compliant with Contributing.md.
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: c/Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+all: $(TARGET)
+
+$(TARGET): $(SRC)
+	$(CC) $(CFLAGS) -o $(TARGET) $(SRC)
+
+clean:
+	rm -f $(TARGET)
+
+check:
+	$(CPPCHECK) $(SRC)
```
Comment (commjoen) [inline c/Makefile:18]: Can you please:
-  add a print out to the check on where to obtain cppcheck for folks that want to use this? 
- update the https://github.com/OWASP/wrongsecrets-binaries/blob/main/quickbuild.sh to include calling this make file instead :) ?

File: c/advanced/Makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+all: $(TARGET)
+
+$(TARGET): $(SRC)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(TARGET) $(SRC)
+
+clean:
+	rm -f $(TARGET)
+
+check:
```
Comment (commjoen) [inline c/advanced/Makefile:18]: Can you please:
-  add a print out to the check on where to obtain cppcheck for folks that want to use this? 
- update the https://github.com/OWASP/wrongsecrets-binaries/blob/main/quickbuild.sh to include calling this make file instead :) ?

File: cplus/Makefile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+all: $(TARGET)
+
+$(TARGET): $(SRC)
+	$(CXX) $(CXXFLAGS) -o $(TARGET) $(SRC)
+
+clean:
+	rm -f $(TARGET)
+
+check:
```
Comment (commjoen) [inline cplus/Makefile:17]: Can you please:
-  add a print out to the check on where to obtain cppcheck for folks that want to use this? 
- update the https://github.com/OWASP/wrongsecrets-binaries/blob/main/quickbuild.sh to include calling this make file instead :) ?

**Review by commjoen — CHANGES_REQUESTED** [review]
Hi @diamant3 !
Thank you for your PR!
Love this! Only some minor feedback :) Can you please have a look and apply the fixes, so we can get you that `hacktoberfest-approved` label :) ?

**Comment by diamant3** [comment]
Thank you for the feedback!. Got it, let me work on that later

**Comment by diamant3** [comment]
Hello, about the dockercross, do I need to update too?

**Comment by commjoen** [comment]
No sir, no need for that, thank you! Can you replace https://github.com/OWASP/wrongsecrets-binaries/blob/main/.github/workflows/compile_c.yml#L24 with your make file call instead ;) ? then we are 🏁

**Comment by commjoen** [comment]
I am also fine if you do that in a separate pr btw, let's roudn this one off ;)

**Comment by commjoen** [comment]
Can you for now, please delete `c/advanced/advanced` on your end and commit the deletion? then we should be good to go. feel free to file another PR with the github action being updated for another hacktoberfest point :-D

**Comment by diamant3** [comment]
Really? Thanks! Okay let me delete that

**Comment by diamant3** [comment]
Sorry, I forgot to remove that, anyway thank you so much for accepting my PR!

**Review by commjoen — APPROVED** [review]
Thank you so much! One more issue down :) .

**Comment by commjoen** [comment]
@diamant3 : can you please let me know your name with which you want to be listed as a contributor?

**Comment by diamant3** [comment]
`diamant3` is fine to me, thank you!

**Comment by commjoen** [comment]
Awesome Diamond  ! will add it soon :) . Can you file the other PR to fix the integration please :) ?

**Comment by diamant3** [comment]
Got it! Okay I will work on that!

**Pull request merged**